### DISCUSSION TITLE
Mount exoress-zod-api in existing express app

### DISCUSSION THREAD

**Discussion by kokotom** [RobinTail/express-zod-api#829] [General]
I want to mount express-zod-api as part of an existing express app?
How can I do it?

When I try to use attachRoutes I am no longer able to access my previous routes

**Comment by RobinTail** [comment]
Hello, @kokotom ,

sorry, I replied you in a sibling discussion about the approach in general:
https://github.com/RobinTail/express-zod-api/discussions/832#discussioncomment-5115146

> When I try to use attachRoutes I am no longer able to access my previous routes

I'm not sure what exactly is happening. Maybe there is an issue on the location in your code where you call thus method.
In general, `attachRouting()` does nothing special but cycles through your `Routing` object and calls `app`'s methods like `.get()` and `post()` for each of the associated Endpoints.
It should not affects on previous calls of these `app`'s methods you might do in your code by yourself.

However, if you have the same route and method assigned twice (by your existing implementation and by utilizing `attachRouting`) then it may lead to the only first of its handlers running.

Anyway. If you could add a code sample and provide more details on exact behavior, I'd probably understand it better. @kokotom
